### PR TITLE
Pdf screen fetching displayed record

### LINKED ISSUE #200: Bring back PDFScreen
Technically the screen is ready to go, but it needs refactoring.

- [x] Refactor PDFScreen to be a bit nicer code-wise (extract functions, name them better, etc)
- [x] Make PDFScreen take the ATR from the display screen on launch, and use that ATR to generate the PDF

For a simple first implementation of the PDF, use `animalTransportRecord.toString()`

### PR DESCRIPTION
**Pull request by SanaKhan1997**
Still needs editing and missing some ATR fields in display. But that can be done with email screen

### CODE REVIEW

File: app/lib/core/models/transporter_info.dart [hunk 1]
```diff
@@ -119,16 +119,16 @@ class TransporterInfo {
...
-  Driver(s) name(s): ${driverNames.join(",")}
-  Province and License Plate number of the conveyance transporting the animals: $vehicleProvince, $vehicleLicensePlate
-  (including trailer): $trailerProvince, $trailerProvince
-  Conveyance or container last cleaned and disinfected date and time: ${DateFormat("yyyy-MM-dd hh:mm").format(dateLastCleaned)}
-  Place: $addressLastCleanedAt
-  Driver(s) have been briefed on the contingency plan?: ${driversAreBriefed ? 'Yes' : 'No'}
-  Driver(s) have received humane transport training?: ${driversHaveTraining ? 'Yes' : 'No'}
-  Training type: $trainingType, Expiry date: ${DateFormat("yyyy-MM-dd hh:mm").format(trainingExpiryDate)}''';
+  String toString() => '''\nName of transporting company: $companyName
+  \nAddress: $companyAddress
```
Comment (holtzmak) [inline app/lib/core/models/transporter_info.dart:123]: Is this working the way you think it is? The new line and space before this `\n` read like they could be included. 

- [x] Could you take a screenshot, or better yet, share the PDF here so we can see?

File: app/lib/ui/widgets/utility/pdf_screen.dart [hunk 2]
```diff
@@ -72,32 +78,83 @@ class _PDFScreenState extends State<PDFScreen> {
...
+            margin: const pw.EdgeInsets.only(top: 1.0 * PdfPageFormat.cm),
+            child: pw.Text(
+              'Page ${context.pageNumber} of ${context.pagesCount}',
+            ));
+      },
       build: (pw.Context context) {
-        /* We will be calling API and fetching data from the database here.
-        Now just adding random text for experimentation purpose. Once we have
-        database/API set up, will experimenting on fetching data */
+        //TODO: Add Complete Atr
```
Comment (holtzmak) [inline app/lib/ui/widgets/utility/pdf_screen.dart]: Is this TODO still valid?

**Review by holtzmak — COMMENTED** [review]
I like the changes so far. Some questions of my own, I'll re-review once you've added ACK images.

**Comment by SanaKhan1997** [comment]
[Test2021-03-23 22_04_54.900.pdf](https://github.com/holtzmak/Humane-Transport/files/6203301/Test2021-03-23.22_04_54.900.pdf)

File: app/lib/ui/widgets/utility/pdf_screen.dart [hunk 3]
```diff
@@ -71,33 +89,128 @@ class _PDFScreenState extends State<PDFScreen> {
...
+      Text("Could not load transporter's  Acknowledgment Receipt");
+    }
+    try {
+      final shipperAckImage =
+          pw.MemoryImage(await networkImageToByte(atr.ackInfo.shipperAck));
+      setState(() {
+        shipperImage = shipperAckImage;
+      });
+    } catch (e) {
+      Text("Could not load transporter's  Acknowledgment Receipt");
```
Comment (holtzmak) [inline app/lib/ui/widgets/utility/pdf_screen.dart]: Having text here, and then not doing anything with it (meaning that the ack image could be null when given to `pw.Image` below) isn't right.

- [x] Do `return Future.error("Could not load transporter's Acknowledgement Receipt")`, etc. instead

File: app/lib/ui/widgets/utility/pdf_screen.dart [hunk 4]
```diff
@@ -71,33 +89,128 @@ class _PDFScreenState extends State<PDFScreen> {
...
       },
     ));
 
-    // TODO: Change the filename to something useful
-    return Future.wait([getTemporaryDirectory(), pdf.save()]).then(
-        (List values) =>
-            File("${values[0].path}/example.pdf").writeAsBytes(values[1]));
+    return Future.wait(
+        [getTemporaryDirectory(), pdf.save()]).then((List values) => File(
+            "${values[0].path}/${atr.deliveryInfo.recInfo.destinationLocationName}${atr.deliveryInfo.arrivalDateAndTime}.pdf")
```
Comment (holtzmak) [inline app/lib/ui/widgets/utility/pdf_screen.dart]: Even though all our PDFs are in the temp directory, there is a possibility the names could collide.

- [x] You could add the ATR document ID in the name again to ensure unique: `${atr.deliveryInfo.recInfo.destinationLocationName}${atr.deliveryInfo.arrivalDateAndTime}${atr.identifier.documentId}`

File: app/lib/ui/widgets/utility/pdf_screen.dart [hunk 5]
```diff
@@ -71,33 +89,128 @@ class _PDFScreenState extends State<PDFScreen> {
...
+                  style: pw.TextStyle(fontSize: 18))),
+          pw.Paragraph(
+              text: widget.atr.deliveryInfo.toString(),
+              style: pw.TextStyle(fontSize: 16)),
+          pw.Header(
+              level: 1,
+              child: pw.Text("Acknowledgements",
+                  style: pw.TextStyle(fontSize: 18))),
+          pw.Text("Transporter's Acknowledgment"),
+          pw.SizedBox(
```
Comment (holtzmak) [inline app/lib/ui/widgets/utility/pdf_screen.dart:193]: Do we not need to limit width to the page?

**Review by holtzmak — CHANGES_REQUESTED** [review]
Minor changes I'd like to see with error handling.

File: app/lib/ui/widgets/utility/pdf_screen.dart [hunk 6]
```diff
@@ -101,7 +108,7 @@ class _PDFScreenState extends State<PDFScreen> {
         tranImage = tranAckImage;
       });
     } catch (e) {
-      Text("Could not load transporter's  Acknowledgment Receipt");
+      Future.error("Could not load transporter's  Acknowledgment Receipt");
```
Comment (holtzmak) [inline app/lib/ui/widgets/utility/pdf_screen.dart]: - [x] Need the return statement!

**Review by holtzmak — CHANGES_REQUESTED** [review]

**Review by holtzmak — APPROVED** [review]

**Pull request merged**